### PR TITLE
Make retry interval configurable

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/10238

Renames the `short-retries` flag to `retry-interval` and allows it to take a configurable value. The units are in minutes, to match the units used by the `reconcile-interval` flag.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Specify new flag as desired (e.g., `--retry-interval=1` to retry once per minute)
- Verify that retries happen as expected per flag

### CODE REVIEW

**Review by zenparsing — APPROVED** [review]

File: components/brave_rewards/browser/test/rewards_flag_browsertest.cc [hunk 1]
```diff
@@ -304,19 +308,19 @@ IN_PROC_BROWSER_TEST_F(RewardsFlagBrowserTest, HandleFlagsWrongInput) {
...
   testing::InSequence s;
   rewards_service_->SetEnvironment(ledger::type::Environment::PRODUCTION);
   rewards_service_->SetDebug(false);
   rewards_service_->SetReconcileInterval(0);
-  rewards_service_->SetShortRetries(false);
+  rewards_service_->SetRetryInterval(0);
 
   rewards_service_->HandleFlags(
-      "staging=,debug=,shortretries=true,reconcile-interval");
+      "staging=,debug=,retryinterval=true,reconcile-interval");
```
Comment (szilardszaloki) [inline components/brave_rewards/browser/test/rewards_flag_browsertest.cc:320]: `retry-interval=1`?

**Review by szilardszaloki — APPROVED** [review]

**Comment by emerick** [comment]
Android/iOS passed; restarting for Mac/Windows/Linux.

**Review by kylehickinson — APPROVED** [review]
iOS lgtm

**Comment by emerick** [comment]
All platforms passed CI; ready to merge.

**Pull request merged**